packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -151,12 +151,11 @@ export class ProfilesCache {
     public async refresh(apiRegister?: extend.IRegisterClient): Promise<void> {
         this.allProfiles = [];
         this.allTypes = [];
-        let mProfileInfo: imperative.ProfileInfo;
         try {
-            mProfileInfo = await this.getProfileInfo();
-            if (!mProfileInfo.usingTeamConfig) {
+            if (imperative.ProfileInfo.onlyV1ProfilesExist) {
```

Comment:
I think the old usingTeamConfig test would return if you only had V1 profiles *OR* you had no configuration of any type. I think your new test will *NOT* return if you have no configuration.

**Alternative 1**
The ZE ProfilesCache class already calls ProfileInfo.readProfilesFromDisk. The return object of that function is a Config object. ZE could add logic to store the Config.exists value, which could then be tested to return from this function only when no configuration exists.

**Alternative 2**
ZE could call ProfileInfo.getTeamConfig.exists for its test.

**Alternative 3**
Perhaps we should expose a property from ProfileInfo (like ProfileInfo.configExists) so that ZE can just test that property when needed? Such a new property could help more consumers. I could add that to the current CLI branch named 'rm-v1-profile-writes' which will eventually be merged into the 'next' branch. Let me know if you think we should add this functionality to the CLI SDK.

Reply:
I do like alternative 3 for a future implementation! 🥳 

I went with option 2 for the time being 😋